src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/resthandler/RestGetTransformActionIT.kt

```diff
@@ -0,0 +1,108 @@
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform.resthandler
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.IndexManagementPlugin.Companion.TRANSFORM_BASE_URI
+import com.amazon.opendistroforelasticsearch.indexmanagement.makeRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.model.dimension.Dimension
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.TransformRestTestCase
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.get.GetTransformsRequest.Companion.DEFAULT_SIZE
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.randomTransform
+import org.elasticsearch.client.ResponseException
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.test.ESTestCase
+import org.elasticsearch.test.junit.annotations.TestLogging
+import org.elasticsearch.test.rest.ESRestTestCase.randomList
+
+@TestLogging(value = "level:DEBUG", reason = "Debugging tests")
+@Suppress("UNCHECKED_CAST")
+class RestGetTransformActionIT : TransformRestTestCase() {
+
+    @Throws(Exception::class)
+    fun `test getting a transform`() {
+        var transform = createTransform(randomTransform())
+        val indexedTransform = getTransform(transform.id)
+
+        transform = transform.copy(
+            schemaVersion = indexedTransform.schemaVersion,
+            updatedAt = indexedTransform.updatedAt,
+            jobSchedule = indexedTransform.jobSchedule
+        )
+        assertEquals("Indexed and retrieved transform differ", transform, indexedTransform)
+    }
+
+    @Throws(Exception::class)
+    fun `test getting a transform that doesn't exist`() {
+        try {
+            getTransform(ESTestCase.randomAlphaOfLength(20))
+            fail("Expected response exception")
+        } catch (e: ResponseException) {
+            assertEquals(RestStatus.NOT_FOUND, e.response.restStatus())
+        }
+    }
+
+    @Throws(Exception::class)
+    fun `test getting all transforms`() {
+        val transforms = randomList(1, 15) { createTransform(randomTransform()) }
```

Comment:
may be add a TODO here to do delete before this test to clear up the existing transforms